iio/iio.c

```diff
@@ -353,30 +367,41 @@ static int iio_read_all_attr(struct attr_fun_params *params,
 static int iio_write_all_attr(struct attr_fun_params *params,
 			      struct iio_attribute *attributes)
 {
-	/* TODO Not sure if working corectly */
-	return -EINVAL;
+	uint32_t read = 0;
 
-#if 0
-	int16_t i = 0, j = 0;
-	int16_t attr_length;
+	/*
+	 * Not all devices have all types of attributes
+	 */
+	if (!attributes)
+		return -ENOENT;
 
-	while (attributes[i].name) {
-		attr_length = bswap_constant_32((uint32_t)(params->buf + j));
-		j += 4;
-		attributes[i].store(params->dev_instance, (params->buf + j),
-				    attr_length, params->ch_info,
-				    attributes[i].priv);
-		j += attr_length;
-		if (j & 0x3)
-			j = ((j >> 2) + 1) << 2;
-		i++;
-	}
+	for (struct iio_attribute *attr = attributes; attr->name; attr++) {
+		uint32_t attr_len;
 
-	if (params->len == 0)
-		return -ENOENT;
+		// Allign to next multiple of 4
+		while (read & 0x3)
```

Comment:
Do we know that data here comes in aligned in this particular way? Do we risk reading the data wrong by doing this?

Reply:
You tell me, the previous code aligned to multiples of 4 too. Is the protocol specification public in documentation other than source code?